content/en/hub/initial-setup/networking/OpenVPN.md

```diff
@@ -0,0 +1,97 @@
+
+### Security Options
+
+Because using a VPN involves connecting to a private network while still sending data over less secure public resources, OpenVPN includes several security options.
+While not required, these security options help protect the data being sent into or out of the private network.
+
+* **Authentication Algorithm**: This is used to validate packets that are sent over the network connection. Your network environment might require a specific algorithm. If no specific algorithm is required, *SHA1 HMAC* is a standard default algorithm.
+* **Cipher**: This is an algorithm to encrypt data packets sent through the connection. While not required, choosing a **Cipher** can increase connection security. You might need to verify which ciphers are required for your networking environment. If there are no specific cipher requirements, *AES-256-GCM* is a good default choice.
+* **TLS Encryption**: When the **TLS Crypt Auth Enabled** checkbox is set, all TLS handshake messages are encrypted to add another layer of security. This requires a static key that is shared between OpenVPN server and clients.
+
+## OpenVPN Server
+
+Go to the **Services** page, find the **OpenVPN Server** row, and click <i class="material-icons" aria-hidden="true" title="Configure">create</i> (Configure) to set up the TrueNAS OpenVPN Server.
+
+<img src="/images/OpenVPNServerServiceConfig.png">
+<br><br>
+
+Choose a **Server Certificate** for this OpenVPN server.
+The certificate must both exist on the TrueNAS system and be in an active (unrevoked) state.
+
+Now define a IP address and netmask for the OpenVPN **Server**.
+Continue to choose the remaining [Connection Settings](#connection-settings) that fit with your network environment and performance requirements.
+When a *TUN* **Device Type** is selected, you can choose a virtual addressing **Topology** for the server:
+
+* *NET30*: Use one `/30` subnet per client in a point-to-point topology. Designed for use when connecting clients are Windows systems.
+* *P2P*: Point-to-point topology that points the local server and remote client endpoints to each other. Each client is given one IP address. This is only recommmended when none of the clients are a Windows system.
+* *SUBNET*: the interface uses an IP address and subnet. Each client is given one IP address. Windows clients require the **TAP-Win32 driver** version 8.2 or newer. *TAP* devices always use the *SUBNET* **Topology**.
+
+The **Topology** selection is automatically applied to any connected clients.
+
+When **TLS Crypt Auth Enabled** is set, TrueNAS generates a static key for the **TLS Crypt Auth** field after saving the options.
+To change this key, click **RENEW STATIC KEY**.
+This key is required for any clients that will connect to the server.
+It is recommended to back up the key in a secure location so that it can be provided to any client systems that will connect to this server.
```

Comment:
Well it's already backed up in the database and if a user wants to give a client configuration to some vpn client, we will add the key to the generated configuration. ( This is just an opinion on how the implementation works, if you think this is good and it's how we advise at other places, i am totally good with it )

Reply:
It's either way for me too - we typically like to tell users to back up keys whenever possible because of the problems that happen when you need the key and don't have it (or don't know where to find it), but this isn't the same level of severity as encrypting data pools. I'll rework a little to include the behavior and keep the general recommendation.